### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by kayibal**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+strategy:
+  matrix:
+    Python27:
+      python.version: '2.7'
+    Python35:
+      python.version: '3.5'
+    Python36:
+      python.version: '3.6'
+    Python37:
+      python.version: '3.7'
```
Comment (carloszanella) [inline azure-pipelines.yml:20]: Like you suggested should we do just 3.6, 3.7 and 3.8?

Comment (kayibal) [reply azure-pipelines.yml:20]: Yeah this PR is autogenerated by Azure ;)

Comment (carloszanella) [reply azure-pipelines.yml:20]: ah ok cool

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+steps:
+- task: UsePythonVersion@0
+  inputs:
+    versionSpec: '$(python.version)'
+  displayName: 'Use Python $(python.version)'
+
+- script: |
+    python -m pip install --upgrade pip
+    pip install -r requirements.txt
```
Comment (giulio-tirabassi) [inline azure-pipelines.yml:30]: requirements_dev is where pytest is specified by the cokiecutter, and where we put dask and loky which are needed to run the test but not to use omigami, so maybe you should add `pip install -r requirements_dev.txt` here ?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+- script: |
+    pip install pytest pytest-azurepipelines==1.0.0rc5
+    pytest --cov=omigami tests/ --cov-report html --nunitxml test-output.xml
+  displayName: 'Run tests'
+
+- script: |
+    make docs
+  displayName: 'Build documentation'
+
+# TODO: these next two are to push to RTD and pypi but the logic is a WIP
```
Comment (giulio-tirabassi) [inline azure-pipelines.yml:38]: the actual push will go in the two script section right?

**Review by giulio-tirabassi — APPROVED** [review]

File: azure-pipelines.yml (hunk 3 above)
Comment (carloszanella) [reply azure-pipelines.yml:38]: exactly

**Pull request merged**